### PR TITLE
feat: update sign up link in share pages

### PR DESCRIPTION
**Pull request by luka-nextcloud**
## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: lib/private/TemplateLayout.php [hunk 1]
```diff
@@ -196,6 +198,14 @@ public function __construct($renderAs, $appId = '') {
 				$showSimpleSignup = false;
 			}
 			$this->assign('showSimpleSignUpLink', $showSimpleSignup);
+
+			$urlGenerator = \OCP\Server::get(IURLGenerator::class);
+			$signUpLink = $this->config->getSystemValueString('registration_link', $urlGenerator->getAbsoluteURL('/index.php/signup/'));
```
Comment (juliushaertl) [inline lib/private/TemplateLayout.php:203]:
```suggestion
			$signUpLink = $this->config->getSystemValueString('registration_link', 'https://nextcloud.com/signup/');
```

File: lib/private/TemplateLayout.php [hunk 2]
```diff
@@ -196,6 +198,14 @@ public function __construct($renderAs, $appId = '') {
...
+			$appManager = \OCP\Server::get(IAppManager::class);
```
Comment (juliushaertl) [inline lib/private/TemplateLayout.php:204]: We should also make sure that `showSimpleSignUpLink` assigned above is still true if a custom URL is configured.

**Review by juliusknorr — CHANGES_REQUESTED** [review]
Two small additions, otherwise this seems fine 👍

**Review by juliusknorr — APPROVED** [review]

File: lib/private/TemplateLayout.php [hunk 3]
```diff
@@ -195,7 +197,21 @@ public function __construct($renderAs, $appId = '') {
 			if ($showSimpleSignup && $subscription->delegateHasValidSubscription()) {
 				$showSimpleSignup = false;
 			}
+
+			$urlGenerator = \OCP\Server::get(IURLGenerator::class);
+			$defaultSignUpLink = 'https://nextcloud.com/signup/';
+			$signUpLink = $this->config->getSystemValueString('registration_link', $defaultSignUpLink);
+			if ($signUpLink !== $defaultSignUpLink) {
+				$showSimpleSignup = true;
```
Comment (icewind1991) [inline lib/private/TemplateLayout.php:205]: do we want to set this back to true if is was set to false on line 198?

Comment (luka-nextcloud) [reply lib/private/TemplateLayout.php:205]: Yes, it should be. As juliushaertl has suggested [here](https://github.com/nextcloud/server/pull/40368/files/e92d396200e5536b2daa3d78c10e594b6ed5538e#r1322043083).

**Review by mejo- — APPROVED** [review]
Looks sensible :ok_hand:

**Review by Altahrim — APPROVED** [review]

**Pull request merged**